cpp/ScanChain_Zp.C

```diff
@@ -94,46 +95,57 @@ int ScanChain(TChain *ch, double genEventSumw, TString year, TString process) {
 
   // Modify the name of the output file to include arguments of ScanChain function (i.e. process, year, etc.)
   TFile* fout = new TFile("temp_data/output_"+process+"_"+year+".root", "RECREATE");
+
   H1(cutflow,20,0,20,"");
-  H1(mll_pf_sel6,240,100,2500,"m_{ll} [GeV]");
-  H1(mll_pf_sel7,240,100,2500,"m_{ll} [GeV]");
-  H1(mll_pf_sel8,240,100,2500,"m_{ll} [GeV]");
-  H1(mll_pf_sel9,240,100,2500,"m_{ll} [GeV]");
-  H1(mu1_pt_sel1,200,0,1000,"p_{T} (leading #mu) [GeV]");
-  H1(mu1_pt_sel2,200,0,1000,"p_{T} (leading #mu) [GeV]");
-  H1(mu1_pt_sel8,200,0,1000,"p_{T} (leading #mu) [GeV]");
-  H1(mu1_pt_sel9,200,0,1000,"p_{T} (leading #mu) [GeV]");
-  H1(mu2_pt_sel1,200,0,1000,"p_{T} (subleading #mu) [GeV]");
-  H1(mu2_pt_sel2,200,0,1000,"p_{T} (subleading #mu) [GeV]");
-  H1(mu2_pt_sel8,200,0,1000,"p_{T} (subleading #mu) [GeV]");
-  H1(mu2_pt_sel9,200,0,1000,"p_{T} (subleading #mu) [GeV]");
-  H1(mu1_trkRelIso_sel2,50,0,0.5,"Track iso./p_{T} (leading #mu)");
-  H1(mu1_trkRelIso_sel3,50,0,0.5,"Track iso./p_{T} (leading #mu)");
-  H1(mu1_trkRelIso_sel8,50,0,0.1,"Track iso./p_{T} (leading #mu)");
-  H1(mu1_trkRelIso_sel9,50,0,0.1,"Track iso./p_{T} (leading #mu)");
-  H1(mu2_trkRelIso_sel2,50,0,0.5,"Track iso./p_{T} (subleading #mu)");
-  H1(mu2_trkRelIso_sel3,50,0,0.5,"Track iso./p_{T} (subleading #mu)");
-  H1(mu2_trkRelIso_sel8,50,0,0.1,"Track iso./p_{T} (subleading #mu)");
-  H1(mu2_trkRelIso_sel9,50,0,0.1,"Track iso./p_{T} (subleading #mu)");
-  H1(nCand_Muons_sel4,4,2,6,"Number of #mu candidates");
-  //H1(btagDeepFlavB_sel6,50,0,1,"b-tag deep-flavor discriminant");
-  H1(nbtagDeepFlavB_sel6,5,0,5,"Number of b-tags (medium WP)");
-  H1(bjet1_pt_sel8,200,0,1000,"p_{T} (leading b-tagged jet) [GeV]");
-  H1(bjet2_pt_sel8,200,0,1000,"p_{T} (subleading b-tagged jet) [GeV]");
-  H1(min_mlb_sel8,200,0,2000,"min m_{lb} [GeV]");
-  H1(min_mlb_sel9,200,0,2000,"min m_{lb} [GeV]");
-  H1(met_pt_sel8,120,0,600,"PF MET p_{T} [GeV]");
-  H1(met_pt_sel9,120,0,600,"PF MET p_{T} [GeV]");
-  H1(met_phi_sel8,65,-3.25,3.25,"PF MET #phi");
-  H1(met_phi_sel9,65,-3.25,3.25,"PF MET #phi");
-  H1(nExtra_muons_sel5,6,0,6,"Number of additional #mu's");
-  H1(nExtra_electrons_sel5,6,0,6,"Number of electrons");
-  H1(nExtra_lepIsoTracks_sel5,6,0,6,"Number of (additional) lepton (e/#mu) PF candidates");
-  H1(nExtra_chhIsoTracks_sel5,6,0,6,"Number of (additional) charged hadron PF candidates");
-  H1(dr_trigobj_sel3,50,0,0.5,"min #Delta R (#mu, trigger object)");
+
+  map<TString, int> nbins { {"mll_pf", 240}, {"mu1_pt", 200}, {"mu2_pt", 200}, {"mu1_eta", 60}, {"mu2_eta", 60}, {"mu1_trkRelIso", 50}, {"mu2_trkRelIso", 50}, {"nCand_Muons", 4}, 
```

Comment:
Here and in the following it may be cleaner to initialize empty maps, then 'insert' entries for each histogram for all maps in block, so that the information for each histogram is kept all together.